### ISSUE TITLE
Document how to install UMS on macOS

### ISSUE THREAD

**Issue by richstokes** [UniversalMediaServer/UniversalMediaServer#1855]
Using v9.0.1 on MacOS Catalina

```
"“Universal Media Server.app” is damaged and can’t be opened. You should move it to the Trash."
```

**Comment by dizer** [comment]
Confirming. As a workaround, you can click on "Open Anyway" in System settings (Security & Privacy)

**Comment by richstokes** [comment]
I dont see that option on latest MacOS. Probably because it thinks the package is "damaged" as opposed to being from an unknown developer?

**Comment by Nadahar** [comment]
The whole "damaged" claim is a red herring. Apple knows exactly what is "wrong" with the file, and that is that the developer didn't pay Apple to issue a "certificate" what would make it perfectly fine. It's strange how easy you can fix things simply by paying Apple.. without actually changing the application.

**Comment by richstokes** [comment]
Well, checking security settings after attempting to open the app I see no option to "open anyway"

**Comment by Nadahar** [comment]
I don't use macOS so I'm not sure what version this started, but they have claimed that unsigned software is "damaged" for many versions now. I think it started around 10.10 - 10.11 somewhere. However, in the earlier versions you could just Apple + click the icon and run it anyway, but that's long gone I think.

**Comment by Nadahar** [comment]
This is related, although this software is signed, but the developer has since stopped paying Apple so the certificate is "expired": https://apple.stackexchange.com/questions/262355/xxx-cant-be-opened-you-should-move-it-to-trash-for-flash-projector-applicat

The cause is the same though.

I think step 4 here still works, although the first 3 are bullshit options IMO: http://osxdaily.com/2019/02/13/fix-app-damaged-cant-be-opened-trash-error-mac/

**Comment by richstokes** [comment]
OK, so this is essentially a "will not fix" situation? I am running UMS in a Linux VM for now, seems kinda crazy but it works!

**Comment by Nadahar** [comment]
"Will not fix" from whom? Apple will not fix it as it is intentional. Free software can not "fix" it since it requires paying Apple money for a "subscription".

**Comment by richstokes** [comment]
Maybe should be phrased "Cannot fix".

How do other FOSS manage this? For example I install a ton of random packages on Mac via homebrew - I'd be very surprised if they have an official Apple certificate but they seem to run OK

**Comment by Nadahar** [comment]
I don't know, maybe homebrew handles this somehow? Maybe those "attributes" are set automatically by homebrew?

**Comment by dizer** [comment]
So I had 7.8 (from long time ago, pre-Catalina for sure) and upgraded to 9.2, which doesn't work. So I reinstalled old 7.8 and got the "Run anyway" button. 

Now, for a test, I tried other versions between 7.8 and 9.2 and none of them was allowed to run. Maybe because I had UMS 7.8 during Catalina update or it just the last working version, not sure.

**Comment by SubJunk** [comment]
> Maybe because I had UMS 7.8 during Catalina update or it just the last working version, not sure.

Can you download a very old version just to check that? There are really old ones at the bottom of https://sourceforge.net/projects/unimediaserver/files/Official%20Releases/OS%20X/

**Comment by dizer** [comment]
Just checked some older versions - only 7.8 which I had before Catalina is allowed to "run anyway"

**Comment by richstokes** [comment]
Sounds like the real question is why does MacOS think the package is "damaged" ? CRC or some other check maybe?

**Comment by Nadahar** [comment]
As said before, it's "damaged" because it isn't signed with an Apple issued developer certificate.

**Comment by richstokes** [comment]
Thats a different error. Security/missing cert error is different from being "damaged" as you can normally hit "run anyway" to bypass that issue.

Same reason it lets you open older versions - something else is going on here

**Comment by Nadahar** [comment]
What is going on is that Apple is gradually tightening the screws. So, they intentionally change error messages from "missing cert" or whatever to "damaged". Have you even tried changing the attributes? If so, I'm pretty sure you'll see that the file will suddenly run just fine.

**Comment by richstokes** [comment]
That works! In which case, can you not ship the app that way?

**Comment by Nadahar** [comment]
I doubt that would work, it would render Apples extortion scheme useless. I'm pretty sure they have found a way to prevent applications setting this, or everybody would just do that instead of paying them.

**Comment by richstokes** [comment]
Maybe ship it with a bash script or something that people can run - with instructions on how to do it. Right now there is no point shipping the Mac version of the app as it wont work for anyone on a modern MacOS version without running the above command :-)

**Comment by Nadahar** [comment]
> Right now there is no point shipping the Mac version of the app as it wont work for anyone on a modern MacOS version without running the above command :-)

This is the case for all software that doesn't pay Apple - it's not unique for UMS. As I said above, I think that in the end the users will have to decide if then want to work around Apples limits, abandon macOS, or simply submit and just use "App store" software.

**Comment by richstokes** [comment]
Fair enough, well thanks for your help.

**Comment by Nadahar** [comment]
This is just my opinion, the UMS developers will have to speak for their stance. I just think it's obvious that anyone that moves beyond the "App store" will have to get used to this.

**Comment by SubJunk** [comment]
I have confirmed I'm getting this bug too.

We are working on a general knowledge base for UMS to consolidate our help resources, that will include that `xattr` script that worked.

I am also interested in other options that don't involve paying Apple, so if you know about any cool tricks/scripts @richstokes I'm all ears :)

**Comment by SubJunk** [comment]
@polygirl I have reassigned this to be a knowledge base article since I think users just need to make sure they cmd+click to open UMS the first time after downloading

**Comment by SubJunk** [comment]
We have also just released 11.3.0 which is has signed and notarized macOS builds, so there should be no more workarounds needed. Hope you enjoy :)
